tests/ga/test_exthandlers_exthandlerinstance.py

```diff
@@ -0,0 +1,95 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the Apache License.
+
+from azurelinuxagent.ga.exthandlers import ExtHandlerInstance
+from azurelinuxagent.common.protocol.restapi import ExtHandler, ExtHandlerProperties
+from tests.tools import *
+
+
+class ExtHandlerInstanceTestCase(AgentTestCase):
+    def setUp(self):
+        AgentTestCase.setUp(self)
+
+        ext_handler_properties = ExtHandlerProperties()
+        ext_handler_properties.version = "1.2.3"
+        ext_handler = ExtHandler(name='foo')
+        ext_handler.properties = ext_handler_properties
+        self.ext_handler_instance = ExtHandlerInstance(ext_handler=ext_handler, protocol=None)
+
+        self.extension_directory = os.path.join(self.tmp_dir, "extension_directory")
+        self.mock_get_base_dir = patch.object(self.ext_handler_instance, "get_base_dir", return_value=self.extension_directory)
+        self.mock_get_base_dir.start()
+
+    def tearDown(self):
+        self.mock_get_base_dir.stop()
+
+    def test_rm_ext_handler_dir_should_remove_the_extension_directory(self):
+        os.mkdir(self.extension_directory)
+        os.mknod(os.path.join(self.extension_directory, "extension_file1"))
+        os.mknod(os.path.join(self.extension_directory, "extension_file2"))
+        os.mknod(os.path.join(self.extension_directory, "extension_file3"))
+
+        self.ext_handler_instance.rm_ext_handler_dir()
+
+        self.assertFalse(os.path.exists(self.extension_directory))
+
+    def test_rm_ext_handler_dir_should_report_an_event_if_an_error_occurrs_while_deleting_the_extension_directory(self):
+        extension_directory = os.path.join(self.tmp_dir, "extension_directory")
+        os.mkdir(extension_directory)
```

Comment:
Why not just: `os.mkdir(self.extension_directory)`, remove the line above and use `self.extension_directory` below?